pkg/operator/certrotationcontroller/externalloadbalancer.go

```diff
@@ -21,7 +21,8 @@ func (c *CertRotationController) syncExternalLoadBalancerHostnames() error {
 		return nil
 	}
 	hostname = strings.Replace(hostname, "https://", "", 1)
-	hostname = hostname[0:strings.LastIndex(hostname, ":")]
+	hostname_arr := strings.Split(hostname, ":")
+	hostname = hostname_arr[0]
```

Comment:
docs say:

```
        // apiServerURL is a valid URI with scheme(http/https), address and
	// port.  apiServerURL can be used by components like the web console
	// to tell users where to find the Kubernetes API.
	APIServerURL string `json:"apiServerURL"`
```

I.e. we are supposed to always have a port. This merely looks like missing validation.

Reply:
@sttts yes based on Doc  it seems missing validation. But why we need to provide port if we are using https scheme with default port 443 ? Is there any dependency or constraints which is forcing to provide ports ?

Reply:
Am fine with this fix here. But I would like to see a counterpart PR fixing the API doc before merging this.

Reply:
@sttts  Thanks
Is this the right doc line which needs to modify?
https://github.com/openshift/api/pull/646

Reply:
yes, file is right. Left a comment.